source/libs/wal/src/walMeta.c

```diff
@@ -201,7 +201,13 @@ static FORCE_INLINE int64_t walScanLogGetLastVer(SWal* pWal, int32_t fileIdx) {
     }
 
     if (end == fileSize) firstTrial = false;
-    if (firstTrial && terrno == TSDB_CODE_SUCCESS) continue;
+    if (firstTrial) {
```

Comment:
If no valid WAL record is detected  in scanning forward while encountering an error, e.g. CHSUM mismatch, then scan backwards. In this time, set firstTrial as false, to prevent it from continuing to scan forwards.